source/common/ur_params.hpp

```diff
@@ -0,0 +1,7780 @@
+/*
+ *
+ * Copyright (C) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * @file ur_params.hpp
+ *
+ */
+#ifndef UR_PARAMS_HPP
+#define UR_PARAMS_HPP 1
+
+#include "ur_api.h"
+#include <ostream>
+
+namespace ur_params {
+
+template <typename T> inline void serializePtr(std::ostream &os, T *ptr);
+
+inline std::ostream &operator<<(std::ostream &os,
```

Comment:
Do you need to add those operators for handles at all? The default implementation for ptrs should work pretty well: it would just print `0` instead of nullptr.

Reply:
Pointers should be printed as 0x... anyway I think, but if you want to have 'nullptr' that's fine

Reply:
With `is_type_complete_v` I was able to remove operators for handles, and instead use serializePtr.